### PR TITLE
DT-1893: ✨ Upgrade to java 16

### PR DESCRIPTION
**Pull request by petergphillips**

### CODE REVIEW

**Review by andrewrlee — DISMISSED** [review]
This is missing circle ci changes?

File: build.gradle.kts [hunk 1]
```diff
@@ -66,6 +66,7 @@ tasks {
   compileKotlin {
     kotlinOptions {
       freeCompilerArgs = listOf("-Xjvm-default=all")
+      jvmTarget = "15"
```
Comment (petergphillips) [inline build.gradle.kts:69]: kotlin doesn't have a target of 16 yet

File: src/test/kotlin/uk/gov/justice/digital/hmpps/oauth2server/auth/repository/UserRepositoryTest.kt [hunk 2]
```diff
@@ -101,9 +101,9 @@ class UserRepositoryTest {
     val retrievedEntity = repository.findByUsername(transientEntity.username).orElseThrow()
 
     // equals only compares the business key columns
-    assertThat(retrievedEntity).usingRecursiveComparison().ignoringFields("authorities").isEqualTo(transientEntity)
-    assertThat(retrievedEntity.username).isEqualTo(transientEntity.username)
-    assertThat(retrievedEntity.email).isEqualTo(transientEntity.email)
+    assertThat(retrievedEntity).usingRecursiveComparison().ignoringFields("authorities", "passwordExpiry", "lastLoggedIn").isEqualTo(transientEntity)
+    assertThat(retrievedEntity.passwordExpiry).isEqualToIgnoringNanos(transientEntity.passwordExpiry)
```
Comment (petergphillips) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/oauth2server/auth/repository/UserRepositoryTest.kt:105]: tests were failing due to nanos after java upgrade

File: .circleci/config.yml [hunk 3]
```diff
@@ -34,13 +37,14 @@ jobs:
...
       name: hmpps/java
-      tag: 11-jdk-browsers
+      tag: 16.0-browsers
     environment:
       _JAVA_OPTIONS: -Xmx512m -XX:ParallelGCThreads=2 -XX:ConcGCThreads=2 -XX:ParallelGCThreads=2 -Djava.util.concurrent.ForkJoinPool.common.parallelism=2 -Dorg.gradle.daemon=false
     steps:
       - checkout
       - attach_workspace:
           at: ~/app
+      - browser-tools/install-browser-tools
```
Comment (petergphillips) [inline .circleci/config.yml:47]: needed as 16.0-browsers doesn't include chrome by default

**Review by SimonMitchellMOJ — APPROVED** [review]
LGTM

**Pull request merged**